doc/classes/Object.xml

```diff
@@ -38,77 +43,119 @@
 			<return type="Variant" />
 			<param index="0" name="property" type="StringName" />
 			<description>
-				Virtual method which can be overridden to customize the return value of [method get].
-				Returns the given property. Returns [code]null[/code] if the [param property] does not exist.
+				Override this method to customize the behavior of [method get]. Should return the given [param property]'s value, or [code]null[/code] if the [param property] does not exist.
 			</description>
 		</method>
 		<method name="_get_property_list" qualifiers="virtual">
 			<return type="Dictionary[]" />
 			<description>
-				Virtual method which can be overridden to customize the return value of [method get_property_list].
-				Returns the object's property list as an [Array] of dictionaries.
-				Each property's [Dictionary] must contain at least [code]name: String[/code] and [code]type: int[/code] (see [enum Variant.Type]) entries. Optionally, it can also include [code]hint: int[/code] (see [enum PropertyHint]), [code]hint_string: String[/code], and [code]usage: int[/code] (see [enum PropertyUsageFlags]).
+				Override this method to customize how script properties should be handled by the engine.
+				Should return a property list, as an [Array] of dictionaries. The result is added to the array of [method get_property_list], and should be formatted in the same way. Each [Dictionary] must at least contain the [code]name[/code] and [code]type[/code] entries.
+				The example below displays [code]hammer_type[/code] in the Inspector dock, and saves it, only if [code]using_hammer[/code] is [code]true[/code]:
+				[codeblock]
+				@tool
+				extends Node2D
+
+				@export var using_hammer = false:
+				    set(value):
+				        using_hammer = value
+				        notify_property_list_changed()
+				var hammer_type = 0
+
+				func _get_property_list():
+				    var properties = []
+
+				    if can_jump:
+				        properties.append({
+				            "name": "hammer_type",
+				            "type": TYPE_INT,
+				            "usage": PROPERTY_USAGE_DEFAULT,
+				            "hint": PROPERTY_HINT_ENUM,
+				            "hint_string": "Wooden,Iron,Golden,Enchanted"
+				        })
+
+				    return properties
+				[/codeblock]
+				[b]Note:[/b] This method is intended for advanced purposes. For most common use cases, the scripting languages offer easier ways to handle properties. See [annotation @GDScript.@export], [annotation @GDScript.@export_enum], [annotation @GDScript.@export_group], etc.
+				[b]Note:[/b] If the object's script is not [annotation @GDScript.@tool], this method will not be called in the editor.
```

Comment:
Love that there's an example showing how this is used!!! Would've been super helpful to me just a few days ago :)

I think line 68 should be `if using_hammer:` rather than `if can_jump:`!

I think it's still useful to have links to Variant.Type, PropertyHint, etc like in the original.

Reply:
I decided not to mention them because all of the information about what this it is present in `get_property_list()`.
I personally believe that all descriptions should be as "self-sustaining" as possible, with very little redirecting around, but this one is an exception because... There's a lot of words.
It's not outstandingly bad to add them back, but it may make the method look more complicated than it *already* is.

Reply:
That's fair, and if people are interested in the enums that do show up like PROPERTY_HINT_ENUM, they can also manually search for them in the documentation anyway!